### ISSUE TITLE
Including Jinja templates from a Django template is broken in Django 1.8

### ISSUE THREAD

**Issue by public** [niwibe/django-jinja#94]
I think this is due to this issue https://code.djangoproject.com/ticket/24425 and specifically this line in 1.8 https://github.com/django/django/blob/8e129b42ad9c9adca599228ae2ad68271fa46d44/django/template/loader_tags.py#L151

This seems to cause the `include` tag to always use the same template engine it started in which prevents django-jinja from doing it's job. Does this need fixing upstream or should django-jinja be implementing some kind of extra smart template engine that can delegate to the DTL backend when necessary?

**Comment by niwibe** [comment]
Sincerely I do not know how answer this question. In my opinion mix two different engines for render a complete template seems a broken approach. In my opinion. If you start rendering with django, it not make sense include jinja engine for it.

At this moment, I'll do not have plans to fix it or help to fix it in django-jinja until have a good reason for mixing engines in rendering templates.

In any case, thanks for report it. It may cause interesting discussion about that.

**Comment by niwibe** [comment]
If django fix that, is ok for me, but I'm pretty sure that if you want include django templates from jinja,  jinja will try render django template with jinja engine.
And I'm not sure that jinja will implement something for solve is. Because the use case of this is slightly questionable.

**Comment by public** [comment]
The problem is that when you have tens of thousands of lines of templates in DTL and you want to port to Jinja having to fix your entire template include hierarchy from the top down is very hard and has very little pay off. For us almost all of the performance advantage tends to come from porting repeatedly included component templates that appear many times on a page.

This means as long as e.g. DTL -> Jinja includes work, things are much easier. It's inconvenient if Jinja -> DTL includes are broken but the scope of the breakage is much reduced. With 1.8 it's particularly frustrating as now our upgrade path is blocked until we port all of our templates or find some other work around to keep the old behaviour :(

Maybe @aaugustin can shed some light on this? :)

**Comment by niwibe** [comment]
Thank @public for your opinion. I understand the reasons. Maybe in django the fix is not very complicated, and if django decision is have support for mix rendering. \

I may will consider port the jinja include tag and make it work in the inverse situation, but the decision depends on the django approach.

**Comment by public** [comment]
Yep. I definitely understand we are using behaviour that may never have been intended here :) I just want to get clarification on exactly how Django and yourself expect this stuff to work before I go and and spend time on another solution.

**Comment by public** [comment]
Would an implementation of the `{% jinja2_include %}` tag for DTL that has been proposed in the upstream bug be welcome as a workaround?

**Comment by aaugustin** [comment]
@niwibe I agree that mixing two different engines for rendering a single page is (at best) a dubious idea. That's why I closed the upstream ticket as wontfix and suggested a custom template tag.

**Comment by niwibe** [comment]
Thanks @aaugustin I'm also agree that maybe the custom templatetag might be a workaround.

**Comment by niwibe** [comment]
I close this Issue due to clear resolution of the problem. Currently I do not have plans work in a specific django_include template tag (extension in jinja language) but I'll gladly accept a pr with it implemented.

**Comment by alanjds** [comment]
I got something working, but had to rename Template.template to Template._template. Will try some PR, maybe breaking existing tests but working for Django 1.11

**Comment by alanjds** [comment]
Got it to work :D

From at least Django 1.7+, the `{% include %}` [accepts](https://github.com/django/django/blob/1.7/django/template/loader_tags.py#L139) a "template-like" having a ``.render()``. So, it is easier to just ``.get_template()`` on a filter and use this one when including:

```python
# myapp/templatetags/get_template.py
from django import template
from django.template import loader

register = template.Library()

@register.filter
def get_template(name):
    return loader.get_template(name)   # <- Uses default template "search" for a engine
```

```django
{# myapp/templates/djangotemplate.html #}
{% load get_template %}
{% include 'myapp/template.jinja'|get_template %}
```

It works up to Django 1.10.X.

On Django 1.11, however, [its internals](https://github.com/django/django/blob/1.11/django/template/loader_tags.py#L207) does some trick searching for ``template.template`` and that breaks django-jinja.

There is a fix on PR #201 to make it work on Django 1.11.
